File: tests/fixtures/sample.ts
/**
 * A sample interface for greeting.
 */
interface Greeter {
    greet(name: string): string;
}

/**
 * A sample class that implements Greeter.
 */
class SampleClass implements Greeter {
    private name: string;

    constructor(name: string) {
        this.name = name;
    }

    greet(name: string): string {
        return `Hello, ${name}`;
    }

    helper(): number {
        return 42;
    }
}

function topLevelFunction(x: number): number {
    return x * 2;
}

File: tests/fixtures/Sample.kt
package com.example

/**
 * A sample interface for greeting.
 */
interface Greeter {
    fun greet(name: String): String
}

/**
 * A sample class that implements Greeter.
 */
class SampleClass(val name: String) : Greeter {
    override fun greet(name: String): String {
        return "Hello, $name"
    }

    fun helper(): Int {
        return 42
    }
}

/**
 * A singleton object.
 */
object Registry {
    fun register(item: String): Boolean {
        return true
    }
}

fun topLevelFunction(x: Int): Int {
    return x * 2
}
